### ISSUE TITLE
[apex] ApexSOQLInjection: False negative when using placeholders via String.replace

### ISSUE THREAD

**Issue by fentes** [pmd/pmd#6388]
**Rule:**[ApexSOQLInjection](https://docs.pmd-code.org/latest/pmd_rules_apex_security.html#apexsoqlinjection)

**Description:** The rule detects the usage of variables by string concatenation. If the query string itself does not contain any variables, it seems to be safe. This allows us to bypass the rule by having placeholders in the query string and replace them afterwards with some dynamic content. 

**Code Sample demonstrating the issue:**

```java
String fieldsString = 'Id';
String objectString = 'Account';
String whereString = 'Id != null';

String queryString = 'SELECT ##fields## FROM ##object## WHERE ##clause##';
queryString = queryString.replace('##fields##', fieldsString);
queryString = queryString.replace('##object##', objectString);
queryString = queryString.replace('##clause##', whereString);

Account testAccount = Database.query(queryString);
```

**Expected outcome:**

PMD should report a violation at line 10, but doesn't as the used variables can contain harmful values.

**Running PMD through:** *IntelliJ AddIn*
